src/quantization.cpp

```diff
@@ -0,0 +1,116 @@
+#include <migraphx/quantization.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/op/convert.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/ranges.hpp>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+instruction_ref insert_fp16(program& prog,
+                            instruction_ref& ins,
+                            shape::type_t type,
+                            std::unordered_map<instruction_ref, instruction_ref>& map_fp16)
+{
+    if(map_fp16.count(ins) > 0)
+    {
+        return map_fp16[ins];
+    }
+
+    assert(ins->get_shape().type() == shape::float_type ||
+           ins->get_shape().type() == shape::double_type);
+    instruction_ref ins_fp16{};
+    if(ins->name() == "@literal" && ins->outputs().size() == 1)
+    {
+        auto l = ins->get_literal();
+        auto s = ins->get_shape();
+        l.visit([&](auto val) {
+            ins_fp16 = prog.add_literal(literal({type, s.lens()}, val.begin(), val.end()));
```

Comment:
With the extended constant propogation we dont need to insert the literals.

Reply:
fixed